### ISSUE TITLE
Attempting to follow a specific users crashes to worker

### ISSUE THREAD

**Issue by dvydra** [jointakahe/takahe#506]
Hi, on my instance running takahe 0.7, I am able to follow most users without issue, however if I follow a specific user the worker process will error and shut down:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 486, in thread_handler
    raise exc_info[1]
  File "/takahe/stator/models.py", line 233, in atransition_attempt
    next_state = await current_state.handler(self)  # type: ignore
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/takahe/users/models/follow.py", line 52, in handle_unrequested
    await follow.source.signed_request(
  File "/takahe/users/models/identity.py", line 915, in signed_request
    return await HttpSignature.signed_request(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/takahe/core/signatures.py", line 251, in signed_request
    raise ValueError(
ValueError: POST error to https://mastodon.cloud/users/coloradotravis/inbox: 401 b'Public key not found for key https://social.ivydra.com/@daniel@social.ivydra.com/#main-key'
Watchdog timeout exceeded
```

**Comment by andrewgodwin** [comment]
Curious - we've done a lot of stator timeout fixes since 0.7, so I can't really act on this as-is. Are you able to upgrade to `main`, or should I put this one aside until we get 0.8 out and you can test that?

(Ignore the specific user thing - it's probably a specific server instead, likely under a lot of pressure and taking ages to respond and thus timing out)

**Comment by dvydra** [comment]
Absolutely, my usage can handle being on main. You can close this and I’ll
let you know if I see it again.

Thanks for the project.

**Comment by dvydra** [comment]
I have never seen this again and I am still following the same accounts. Happy to close as fixed since 0.8

**Comment by andrewgodwin** [comment]
Hooray!
